src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Cohost/HtmlRequestInvoker.cs

```diff
@@ -91,15 +84,35 @@ internal sealed class HtmlRequestInvoker(
         }
         finally
         {
-            Debug.Assert(
-                (textDocumentRequest is null && originalUri is null) ||
-                (textDocumentRequest is not null && originalUri is not null), "textDocumentRequest and originalUri should either both be null or both be non-null");
-
-            // Reset the Uri if we changed it.
-            if (textDocumentRequest is not null && originalUri is not null)
+            if (originalUri is not null)
             {
-                textDocumentRequest.TextDocument.DocumentUri = originalUri;
+                UpdateTextDocumentUri(request, originalUri, out _);
             }
         }
     }
+
+    private void UpdateTextDocumentUri<TRequest>(TRequest request, DocumentUri uri, out DocumentUri? originalUri) where TRequest : notnull
+    {
+        switch (request)
```

Comment:
>switch (request) [](http://example.com/codeflow?start=8&length=16)

nit: what do you think of:

```
        var textDocument = request switch
        {
            ITextDocumentParams textDocumentParams => textDocumentParams.TextDocument,
            VSInternalDiagnosticParams vsInternalDiagnosticParams => vsInternalDiagnosticParams.TextDocument,
            VSInternalRelatedDocumentParams vsInternalRelatedDocumentParams => vsInternalRelatedDocumentParams.TextDocument,
            _ => null
        };

        originalUri = textDocument?.DocumentUri;
        textDocument?.DocumentUri = uri;
```

Reply:
Derp, yep, that makes perfect sense. I original wrote the switch without capturing the textdocument, so each arm operated on a different type. Should have thought to go back and review after. Thanks.